idos-build/docker-run.sh

```diff
@@ -13,32 +13,69 @@ socat -d -d TCP4-LISTEN:443,fork,bind=127.0.0.4 VSOCK-CONNECT:3:6010 &
 socat -d -d TCP4-LISTEN:443,fork,bind=127.0.0.5 VSOCK-CONNECT:3:6011 &
 socat -d -d TCP4-LISTEN:443,fork,bind=127.0.0.6 VSOCK-CONNECT:3:6012 &
 socat -d -d TCP4-LISTEN:443,fork,bind=127.0.0.7 VSOCK-CONNECT:3:6013 &
+socat -d -d TCP4-LISTEN:443,fork,bind=127.0.0.8 VSOCK-CONNECT:3:6014 &
+socat -d -d TCP4-LISTEN:443,fork,bind=127.0.0.9 VSOCK-CONNECT:3:6015 &
+socat -d -d TCP4-LISTEN:443,fork,bind=127.0.0.10 VSOCK-CONNECT:3:6016 &
 
 # So that the FaceTec .so file can load some stuff on /tmp.
 mount /tmp -o remount,exec
 
-# Mount s3fs
-S3_BUCKET="nitro-enclave-usage-logs"
-MOUNT_POINT="/home/FaceTec_Custom_Server/deploy/facetec_usage_logs_server/facetec-usage-logs"
-mkdir -p "$MOUNT_POINT"
-echo "Making sure $MOUNT_POINT is un-mounted"
-umount "$MOUNT_POINT" || true
-echo "Mounting $MOUNT_POINT"
-s3fs "$S3_BUCKET" "$MOUNT_POINT" -o iam_role=auto
-echo "Done with mounting $MOUNT_POINT"
-
-S3_3D_DB_BUCKET="nitro-enclave-3d-db"
-MOUNT_POINT_3D_DB="/home/FaceTec_Custom_Server/deploy/three_d_db"
-mkdir -p "$MOUNT_POINT_3D_DB"
-echo "Making sure $MOUNT_POINT_3D_DB is un-mounted"
-umount "$MOUNT_POINT_3D_DB" || true
-echo "Mounting $MOUNT_POINT_3D_DB"
-s3fs "$S3_3D_DB_BUCKET" "$MOUNT_POINT_3D_DB" -o iam_role=auto
-echo "Done with mounting $MOUNT_POINT_3D_DB"
-
-cd /home/FaceTec_Custom_Server/deploy/facetec_usage_logs_server
-npm start
-echo '# ULS done, going for java #'
-
-cd ..
-java -jar FaceTec-Custom-Server.jar
+# Helpers
+mount_s3() {
+  local bucket="$1"
+  local mount_point="$2"
+  mkdir -p "$mount_point"
+  echo "Ensuring $mount_point is unmounted"
+  umount "$mount_point" || true
+  echo "Mounting S3 bucket $bucket to $mount_point"
+  s3fs "$bucket" "$mount_point" -o iam_role=auto
+}
+
+mount_gocryptfs() {
+  local s3_folder="$1"
+  local mount_point="$2"
+  local key_file="$3"
+
+  mkdir -p "$mount_point"
+  echo "Ensuring $mount_point is unmounted"
+  umount "$mount_point" || true
+
+  if [ ! -e "$s3_folder/gocryptfs.conf" ]; then
+    echo "Initializing gocryptfs in $s3_folder"
+    sops -d "$key_file" | gocryptfs -init "$s3_folder" -nosyslog
+  else
+    echo "$s3_folder already initialized"
+  fi
+
+  echo "Mounting gocryptfs $s3_folder -> $mount_point"
+  sops -d "$key_file" | gocryptfs "$s3_folder" "$mount_point" -nosyslog
+}
+
+# Mount the keys first
+KEY_MOUNT_POINT="/home/FaceTec_Custom_Server/deploy/keys"
+mount_s3 "nitro-enclave-hello-config" "$KEY_MOUNT_POINT"
```

Comment:
This is a bit confusing for me. Why not store the yaml files alongside the conf files?

Reply:
We talked about this over voice. Even though my idea might work, we don't want to go through the trouble of discovering how well gocryptfs would deal with having a file places manually in a dir that only gocryptfs should manage.